test/extended/builds/run_policy.go

```diff
@@ -0,0 +1,222 @@
+package builds
+
+import (
+	"fmt"
+	"strings"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	buildclient "github.com/openshift/origin/pkg/build/client"
+	buildutil "github.com/openshift/origin/pkg/build/util"
+	exutil "github.com/openshift/origin/test/extended/util"
+	kapi "k8s.io/kubernetes/pkg/api"
+)
+
+var _ = g.Describe("[builds][Slow] using build configuration runPolicy", func() {
+	defer g.GinkgoRecover()
+	var (
+		// Use invalid source here as we don't care about the result
+		oc = exutil.NewCLI("cli-build-run-policy", exutil.KubeConfigPath())
+	)
+
+	g.JustBeforeEach(func() {
+		g.By("waiting for builder service account")
+		err := exutil.WaitForBuilderAccount(oc.KubeREST().ServiceAccounts(oc.Namespace()))
+		o.Expect(err).NotTo(o.HaveOccurred())
+		// Create all fixtures
+		oc.Run("create").Args("-f", exutil.FixturePath("..", "extended", "fixtures", "run_policy")).Execute()
+	})
+
+	g.Describe("build configuration with Parallel build run policy", func() {
+		g.It("runs the builds in parallel", func() {
+			g.By("starting multiple builds")
+			var (
+				startedBuilds []string
+				counter       int
+			)
+			bcName := "sample-parallel-build"
+
+			buildWatch, err := oc.REST().Builds(oc.Namespace()).Watch(kapi.ListOptions{
+				LabelSelector: buildutil.BuildConfigSelector(bcName),
+			})
+			defer buildWatch.Stop()
+
+			// Start first build
+			out, err := oc.Run("start-build").Args(bcName).Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			startedBuilds = append(startedBuilds, strings.TrimSpace(out))
+
+			// Wait for it to become running
+			for {
+				event := <-buildWatch.ResultChan()
+				build := event.Object.(*buildapi.Build)
+				if build.Name == startedBuilds[0] && build.Status.Phase == buildapi.BuildPhaseRunning {
+					break
+				}
+			}
+
+			for i := 0; i < 2; i++ {
+				out, err := oc.Run("start-build").Args(bcName).Output()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				startedBuilds = append(startedBuilds, strings.TrimSpace(out))
+			}
+
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			// Count pending builds here and if all builds we started are in Pending,
+			// check their state again.
+			for {
+				event := <-buildWatch.ResultChan()
+				build := event.Object.(*buildapi.Build)
+				if build.Name == startedBuilds[0] {
+					if buildutil.IsBuildComplete(build) {
+						break
+					}
+					continue
+				}
+				if build.Status.Phase == buildapi.BuildPhasePending {
+					counter++
+				}
+				if counter == 2 {
+					break
+				}
+			}
```

Comment:
i reworked that test to check for the state of the builds... if they are completed we exit immediately with error... that means the only timeout we can hit here is when the build takes ages to become pending (which I don't think happens very often).

The counter is still there to know when to exit the loop, but it has no testing value.